src/main/scala/org/broadinstitute/hail/Utils.scala

```diff
@@ -596,17 +598,17 @@ class RichPairTraversableOnce[K, V](val t: TraversableOnce[(K, V)]) extends AnyV
   }
 }
 
-class RichPairIterator[K, V](val it: Iterator[(K, V)]) {
+class RichPairIterator[K : Ordering, V](val it: Iterator[(K, V)]) {
```

Comment:
I'm bothered by this class because it is the only one which escaped the transformation to relying on `OrderedKey[PK,K]`. In particular, the ordering chosen for `K` can theoretically disagree with the ordering for `K` given by `OrderedKey[PK,K]`.

I attempted to add an implicit argument for `OrderedKey[PK,K]` to this class:

```scala
class RichPairIterator[PK, K, V](val it: Iterator[(K,V)])(implicit kOk: OrderedKey[PK,K]) // ...
```

Unfortunately this kills the implicit conversion engine. In particular, the value of `PK` is always ambiguous and the engine does not try to choose a `PK` by looking for a unique type of the shape `OrderedKey[_,K]`. Honestly, this is probably the right choice for Scala.

I'm tempted to make these static methods which explicitly take an `OrderedKey[PK,K]`. I suppose this looks ugly though, right?

cc @cseed